llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -2689,6 +2689,12 @@ def int_amdgcn_global_load_tr_b128 : AMDGPULoadIntrinsic<global_ptr_ty>;
 def int_amdgcn_wave_id :
   DefaultAttrsIntrinsic<[llvm_i32_ty], [], [NoUndef<RetIndex>, IntrNoMem, IntrSpeculatable]>;
 
+def int_amdgcn_s_prefetch_data :
+  Intrinsic<[], [llvm_anyptr_ty, llvm_i32_ty],
```

Comment:
Comment the operands? How is this different from the regular llvm.prefetch?

Reply:
Added comments. I will have to guard the regular llvm.prefetch, the instruction turns to be unsafe with invalid address. Besides target intrinsic can take length operand.